src/permission-set.js

```diff
@@ -16,523 +16,809 @@
+     * (If true, uses `options.origin` parameter)
+     * @property strictOrigin
+     * @type {Boolean}
+     */
+    this.strictOrigin = options.strictOrigin
+    /**
+     * Contents of the request's `Origin:` header.
+     * (used only if `strictOrigin` parameter is set to true)
+     * @property origin
+     * @type {String}
+     */
+    this.origin = options.origin
+    /**
+     * Solid REST client (optionally injected), used by save() and clear().
+     * @type {SolidWebClient}
+     */
+    this.webClient = options.webClient
+
+    // Init the functions for deriving an ACL url for a given resource
+    this.aclUrlFor = options.aclUrlFor
+      ? options.aclUrlFor
+      : PermissionSet.defaultAclUrlFor
+    this.aclUrlFor.bind(this)
+    this.isAcl = options.isAcl ? options.isAcl : PermissionSet.defaultIsAcl
+    this.isAcl.bind(this)
+
+    // Optionally initialize from a given parsed graph
+    if (options.graph) {
+      this.initFromGraph(options.graph)
+    }
+  }
+
   /**
-   * Hashmap of all Authorizations in this permission set, keyed by a hashed
-   * combination of an agent's/group's webId and the resourceUrl.
-   * @property authorizations
-   * @type {Object}
+   * Returns the corresponding ACL uri, for a given resource.
+   * This is the default `aclUrlFor()` method that's used by
+   * PermissionSet instances, unless it's overridden in options.
+   * @method defaultAclUrlFor
+   * @param resourceUri {String}
+   * @return {String} ACL uri
    */
-  this.authorizations = {}
+  static defaultAclUrlFor (resourceUri) {
+    if (this.isAcl(resourceUri)) {
+      return resourceUri  // .acl resources are their own ACLs
+    } else {
+      return resourceUri + this.aclSuffix
+    }
+  }
+
   /**
-   * The URL of the corresponding ACL resource, at which these permissions will
-   * be saved.
-   * @property aclUrl
-   * @type {String}
+   * Tests whether a given uri is for an ACL resource.
+   * This is the default `isAcl()` method that's used by
+   * PermissionSet instances, unless it's overridden in options.
+   * @method defaultIsAcl
+   * @param uri {String}
+   * @returns {Boolean}
    */
-  this.aclUrl = aclUrl
+  static defaultIsAcl (uri) {
+    return uri.endsWith(this.aclSuffix)
+  }
+
   /**
-   * RDF Library (optionally injected)
-   * @property rdf
-   * @type {RDF}
+   * Adds a given Authorization instance to the permission set.
+   * Low-level function, clients should use `addPermission()` instead, in most
+   * cases.
+   * @method addAuthorization
+   * @private
+   * @param auth {Authorization}
+   * @return {PermissionSet} Returns self (chainable)
    */
-  this.rdf = options.rdf
+  addAuthorization (auth) {
+    var hashFragment = auth.hashFragment()
+    if (hashFragment in this.authorizations) {
+      // An authorization for this agent and resource combination already exists
+      // Merge the incoming access modes with its existing ones
+      this.authorizations[ hashFragment ].mergeWith(auth)
```

Comment:
I don't have a personal preference as to whether or not we should pad `[` with spaces, but I'm surprised that `standard` isn't complaining.